src/Services/PluginInfoService.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace Findologic\Services;
+
+use Plenty\Modules\Plugin\Contracts\PluginRepositoryContract;
+use Plenty\Modules\Plugin\Models\Plugin;
+use Plenty\Modules\Plugin\PluginSet\Contracts\PluginSetRepositoryContract;
+use Plenty\Plugin\CachingRepository;
+
+class PluginInfoService
+{
+    const PLUGIN_VERSION_CACHE_KEY_PREFIX = 'plugin_version_';
+
+    /**
+     * @var PluginRepositoryContract|null
+     */
+    private $pluginRepository;
+
+    /**
+     * @var CachingRepository
+     */
+    private $cache;
+
+    /**
+     * @var Plugin|null[]
```

Comment:
That typehint is either wrong or it is used in a wrong way. It is supposed to store an array of plugins, but the typehint either contains a single plugin or an array of null, which doesn't really make sense. I think the proper typehint should be `Plugin[]`.